### PR TITLE
Remove deprecated factories usage

### PR DESCRIPTION
**Pull request by kennyadsl**
This PR is part of [3.x Deprecations Removal](https://github.com/solidusio/solidus/issues/4983).

## Summary

This PR removes support for requiring individual factories from Solidus core using: 

```
require 'spree/testing_support/factories/some_factory'
```

Now, you should use the provided `FactoryBot.definition_file_paths`. We opened this [GitHub Discussion](https://github.com/solidusio/solidus/discussions/4014) with more details, and instructions to upgrade.

Ref https://github.com/solidusio/solidus/pull/3907

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [ ] I have written a thorough PR description.
- [ ] I have kept my commits small and atomic.
- [ ] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

### CODE REVIEW

File: core/lib/spree/testing_support/factory_bot.rb [hunk 1]
```diff
@@ -17,44 +17,6 @@ def self.definition_file_paths
...
-        requirement = Gem::Requirement.new(">= 4.8")
-        version = Gem::Version.new(::FactoryBot::VERSION)
-
-        unless requirement.satisfied_by? version
-          Spree::Deprecation.warn(
-            "Please be aware that the supported version of FactoryBot is #{requirement}, " \
-            "using version #{version} could lead to factory loading issues.", caller(2)
-          )
-        end
-      end
```
Comment (elia) [inline core/lib/spree/testing_support/factory_bot.rb:56]: Should we turn this into a hard failure for versions below 4.8?

**Review by elia — APPROVED** [review]
❤️

File: core/lib/spree/testing_support/factory_bot.rb (hunk 1 above)
Comment (waiting-for-dev) [reply core/lib/spree/testing_support/factory_bot.rb:56]: :+1:

**Review by waiting-for-dev — APPROVED** [review]

File: core/lib/spree/testing_support/factory_bot.rb (hunk 1 above)
Comment (waiting-for-dev) [reply core/lib/spree/testing_support/factory_bot.rb]: Done!

**Review by elia — APPROVED** [review]

**Pull request merged**